app/models/job-link.ts

```diff
@@ -0,0 +1,162 @@
+import _ from 'lodash';
+import { Transaction } from 'util/db';
+import { removeEmptyProperties } from 'util/object';
+import Record from './record';
+
+interface BaseJobLink {
+  id?: number;
+  jobID?: string;
+  href: string;
+  type?: string;
+  title?: string;
+  rel?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+interface JobLinkRecord extends BaseJobLink {
+  bbox?: string;
+  temporalStart?: Date;
+  temporalEnd?: Date;
+}
+
+/**
+ * For the constructor we can take either a JobLink or a JobLinkRecord.
+ * When created from the database it will be a JobLinkRecord, but when
+ * constructed outside of the database it will likely be a JobLink
+ * which has a different representation for the bbox and temporal.
+ * Specifying JobLink | JobLinkRecord as the type causes many Typescript
+ * errors so we use this interface to avoid that.
+ */
+
+export interface JobLinkOrRecord extends BaseJobLink {
+  bbox?: string | number[];
+  temporalStart?: Date;
+  temporalEnd?: Date;
+  temporal?: {
```

Comment:
Why have this and `temporalStart`/`temporalEnd`?

Reply:
This is the point of the interface - you can either pass in the temporal JSON with start and end properties or temporalStart and temporalEnd. In most places we use the temporal JSON, but in the database we are storing temporalStart and temporalEnd as two separate timestamps.